tiledb/sm/rest/rest_profile.cc

```diff
@@ -179,8 +189,9 @@ void RestProfile::save_to_file() {
         "either both be set or both remain unset. Mixing a default username "
         "with a custom password (or vice versa) is not allowed.");
   }
-  // Fstream cannot create directories. If `homedir/.tiledb/` DNE, create it.
-  std::filesystem::create_directories(homedir_ + ".tiledb");
+  // Fstream cannot create directories. If `profile_dir/.tiledb/` DNE, create
+  // it.
+  std::filesystem::create_directories(dir_ + ".tiledb");
```

Comment:
THought I'd check, and it *is* specified that this creates parent directories - glad we don't have to do that manually!

I've expressed before that I have gripes with appending `.tiledb` to whatever the user-supplied path is... don't change it in this PR but I prefer that be done before this feature is released